Library/Homebrew/formula_auditor.rb

```diff
@@ -339,6 +339,18 @@ def audit_postgresql
       end
     end
 
+    def audit_glibc
+      return if formula.name != "glibc"
+      return unless @core_tap
+
+      version = formula.version.to_s
+      return if version == "2.23"
+
+      problem "The glibc version can only be #{version} as this is the version used by our CI on Linux. " \
+              "Glibc is only there for users that have a system glibc with a lower version, " \
+              "which allows them to use our linux bottles (which were compiled against system glibc on CI)."
```

Comment:
How can this be upgraded? CI will fail if this version ever changes.

Reply:
I would set `return if version <= "2.27"` (if you want to bump to version 2.27 for example).
This does not break the old formula and allows to upgrade in a controlled way.

Reply:
So it would require Homebrew/brew code changes merged first in order to upgrade it? I'm a bit tetchy about a solution that requires a change here before an upgrade; is there any reason we couldn't do something like that now?

Reply:
Upgrading glibc to a newer version also means updating the CI docker images, from ubuntu 16.04 to 18.04 for example + updating the docs + dropping support for older versions. And even like that you would not be on the latest glibc version. I plan to do that migration (maybe during this summer), but it involves updating the OS, the minimum glibc version and transitioning from gcc 5 to gcc 7. It's not a small task.

Another solution is to add 2 formulae, glibc@2.23 and glibc (which would be 2.33 == the latest version). Nothing would be allowed to depend on glibc, only glibc@2.23 would be allowed in formulae.

Yet another solution is to drop support for older systems (pre-ubuntu 16.04, with glibc between 2.13 and 2.23), which means we would probably not need a glibc formula. This means that in general we would be less supportive for old operating systems.

Reply:
So anything that touches glibc is better handled by a Linux maintainer that has some background knowledge about the topic, that's why the audit is better in brew, to prevent developers to bump glibc to any random version in homebrew-core.

Reply:
> Another solution is to add 2 formulae, glibc@2.23 and glibc (which would be 2.33 == the latest version). Nothing would be allowed to depend on glibc, only glibc@2.23 would be allowed in formulae.

This would also be fine with me, I have no strong feelings either way.



> So anything that touches glibc is better handled by a Linux maintainer that has some background knowledge about the topic, that's why the audit is better in brew, to prevent developers to bump glibc to any random version in homebrew-core.

How about a `CODEOWNERS` rule which requests review from Homebrew/linux and/or you? That's what we did for GitHub Actions? That combined with formulae comments seem like it'd be enough without an `audit`? I'm fine either way, though, ✅

Reply:
> How about a CODEOWNERS rule which requests review from Homebrew/linux and/or you? That's what we did for GitHub Actions?

We could do that too. Though it I think it does not give an immediate explanation of why there is a codeowner on that file.

Let's ship this, it should be the only formula which has a pinned version, so I think this audit is enough right now.